src/ec/suite_b/ecdsa/signing.rs

```diff
@@ -163,6 +165,23 @@ impl EcdsaKeyPair {
         self.sign_(rng, h)
     }
 
+    /// Returns the signature of the `digest` using a random nonce
+    /// generated by `rng`.
+    ///
+    /// The `digest` algorithm must match that of the signing algorithm.
+    pub fn sign_digest(
```

Comment:
Please name this `sign_digest_less_safe`.

Please add a note like the following to the documentation: "In general, it is not safe to sign an arbitrary digest. Ensure that you only sign digests that you have computed yourself, or that you otherwise know are safe to sign. It could be a bad mistake to sign an attacker-controlled digest."